### ISSUE TITLE
[Replace] spaces

### ISSUE THREAD

**Issue by GraineDeMoutarde** [erosman/support#130]
Hi, i'd like to replace black spaces like this '          ' with only one space like this ' ' on a web page and it doesn't work with the tool replace. Is there any way so that it recognizes spaces ?

Thank you

**Comment by erosman** [comment]
Can you give me an example page to test?

**Comment by GraineDeMoutarde** [comment]
here is an example https://mensuel.framapad.org/p/9e5m-pagedaccueil?lang=fr
you have 5 spaces between each word and there should be only one

**Comment by erosman** [comment]
Thank you. I will work on it.

**Comment by GraineDeMoutarde** [comment]
Well, I've searched the web for ideas and i found that there could be a command that does the job (like "^w" is used on Word for spaces (but it doesn't work with Replace)).

**Comment by erosman** [comment]
Actually, the code at the moment doesn't allow only spaces. It checks if FIND has anything other than space in it. Obviously, it won't work with only space now.

I have changed that so I will accept spaces.

Now, it will remove normal spaces. However, in your example, the space are **Non-breaking space**

`L'zvenir&nbsp;&nbsp;&nbsp;&nbsp; ce&nbsp;&nbsp;&nbsp;&nbsp; n'est&nbsp;&nbsp;&nbsp;&nbsp; pzs&nbsp;&nbsp;&nbsp;&nbsp; ce&nbsp;&nbsp;&nbsp;&nbsp;`

I have to see what need to be done to remove those.

**Comment by erosman** [comment]
I forgot to mention that you can use RegEx (Regular Expression) which works e.g.: `/\s+/` means multiple spaces.

**Comment by GraineDeMoutarde** [comment]
Ok i've used the tool to reduce mutliple spaces to one on my non-example page, and it worked on some of the blanks. Some are still remaining probably for the reason you mentionned that they are non-breaking spaces.

If you want you can check this page at https://semestriel.framapad.org/p/francaispcsi3?lang=fr. I'm just afraid some other people might be working on it (for its content, not for the its technical problems) at the same time but it happends very seldom... 

This page has been fulled with a pasted content and there are lots of blanks everywhere. I've corrected them manually at the beginning of the text so you will find them more at the end (especially where there are bullet points). I think there could be some other stuff than non breaking space like indentation (i've tried some from a Redex documentation but none worked).

However this first wave of space reduction has bettered it a lot, thank you very much for helping !

**Comment by erosman** [comment]
> If you want you can check this page at https://semestriel.framapad.org/p/francaispcsi3?lang=fr. 

Can you copy/paste the doc somewhere else for testing since edits are live and I dont want to mess up the page?

**Comment by GraineDeMoutarde** [comment]
good idea ! it's on the example pad https://mensuel.framapad.org/p/9e5m-pagedaccueil?lang=fr

**Comment by erosman** [comment]
> In order to reduce spaces to one /\s\s+/ and replace with `'  '`

Above clears most of them except places that spaces are without any text like blank list lines.

`/\u00A0(\u00A0)+/` and replace with `'  '` also removes some but not all.

The editable document is a whole different types of document where its script is running to create that document format. I will do some more testing.

**Comment by GraineDeMoutarde** [comment]
Hi ! Have you found other shortcuts that could help cleaning the page ? Else with the one you've given me it already does a big part of the job, thank you for it !

**Comment by erosman** [comment]
> Have you found other shortcuts that could help cleaning the page ?

Sorry, not yet. I will let you know if I find any.
